### DISCUSSION TITLE
Named Lists of Names

### DISCUSSION THREAD

**Discussion by mo-nathan** [MushroomObserver/mushroom-observer#1898] [Ideas]
As part of assessing the requirements for the 2024 Northeast Mycological Federation foray, I have identified a couple of interesting use cases for persistent lists of names that are different from the current MO Species List concept.  The use cases are:

1) Review a list of names like those from the NEMF name list used by their recorder.
2) Provide a way to store and revise the set of names associated with a location in a foray id process (e.g., Table 1 has Amanitaceae).

To elaborate on the first case, the idea is to take a list of name strings that may or may not be in MO.  Then identify which correspond to existing names and provide links to those names.  If those corresponding names are deprecated, show the currently preferred name (or names) in MO.  If they are not currently in MO, indicate that and provide an easy way to add them.  It would also be desirable to automatically provide links to the name in various well known resources like Index Fungorum, MycoBank, iNat, Google Search, etc.

The second case is more specialized for foray id processes.  However, this does seem to be a common and useful function for the systems in use and mushroom events.  Ideally we'd create a small list of names (E.g., Cortinariaceae, Inocybaceae) and then be able retrieve information like "Table 10" from an included taxon like "Inocybe abjecta".  In thinking this mechanism would be part of Projects.  So a Project would have a set of named lists of names (e.g., "Table 10").  Then when you look at an observation associated with that Project, it could show you which of the lists that name is part of.  I recognize that this is likely an expensive query, so it wouldn't go on the Show Observation page.  Maybe this goes on the Project name checklist page?  E.g., https://mushroomobserver.org/checklist?project_id=336.  It would be ideal if there was a quick way to go from the planned Voucher page to this info.  That way a "runner" could scan the QR code, and quickly find the associated named list.

**Comment by JoeCohen** [comment]
For use case 1, could you please implement this 2 ago. 😉
I need to get my head around use case 2 and understand the physical work flow better. It sounds like a runner looks up a name in the List of Names, and gets a response like "Table 3, Table 10". Is that right? Is the goal that the runner knows where to put the specimen? or is it where to look for the specimen?

**Reply by mo-nathan** [reply]
The use case is based on letting the runner know where the specimen goes.  Using to figure out where to look for a species is interesting.  There is also an interesting question of what happens if a name applies to more than one list.  I think you'd want a unique resolution.  I guess you'd have to figure out the set of lists that it applies to, then at what level the match (a match at the species levels is a better match than one at the family level).  If it's still ambiguous, maybe pick the "first" one (implies the tables are ordered).

Hmm, now that I think about it more, maybe the lists just have genera or explicit species.  If there isn't a match then the person asks and the name gets added to one of the lists.  The reason for the explicit species might be a table for rare species, choice edibles, or poisonous species.
